--- ../src-base/minecraft/net/minecraft/block/BlockTripWire.java
+++ ../src-work/minecraft/net/minecraft/block/BlockTripWire.java
@@ -42,6 +42,11 @@
         this.func_149675_a(true);
     }
 
+    @Override
+    public int func_149738_a(World worldIn) {
+        return 1;
+    }
+
     public AxisAlignedBB func_185496_a(IBlockState p_185496_1_, IBlockAccess p_185496_2_, BlockPos p_185496_3_)
     {
         return !((Boolean)p_185496_1_.func_177229_b(field_176294_M)).booleanValue() ? field_185748_C : field_185747_B;
