--- ../src-base/minecraft/net/minecraft/util/math/MathHelper.java
+++ ../src-work/minecraft/net/minecraft/util/math/MathHelper.java
@@ -405,6 +405,19 @@
         return p_188208_0_;
     }
 
+
+    public static long getLong(String value, long defaultValue)
+    {
+        try
+        {
+            return Long.parseLong(value);
+        }
+        catch (Throwable var3)
+        {
+            return defaultValue;
+        }
+    }
+    
     static
     {
         for (int i = 0; i < 65536; ++i)
