--- ../src-base/minecraft/net/minecraft/network/play/server/SPacketUpdateScore.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/SPacketUpdateScore.java
@@ -11,7 +11,7 @@
 {
     private String field_149329_a = "";
     private String field_149327_b = "";
-    private int field_149328_c;
+    private long field_149328_c;
     private SPacketUpdateScore.Action field_149326_d;
 
     public SPacketUpdateScore()
@@ -22,7 +22,7 @@
     {
         this.field_149329_a = p_i46904_1_.func_96653_e();
         this.field_149327_b = p_i46904_1_.func_96645_d().func_96679_b();
-        this.field_149328_c = p_i46904_1_.func_96652_c();
+        this.field_149328_c = p_i46904_1_.getScorePoints();
         this.field_149326_d = SPacketUpdateScore.Action.CHANGE;
     }
 
@@ -50,7 +50,7 @@
 
         if (this.field_149326_d != SPacketUpdateScore.Action.REMOVE)
         {
-            this.field_149328_c = p_148837_1_.func_150792_a();
+            this.field_149328_c = p_148837_1_.readLong();
         }
     }
 
@@ -62,7 +62,7 @@
 
         if (this.field_149326_d != SPacketUpdateScore.Action.REMOVE)
         {
-            p_148840_1_.func_150787_b(this.field_149328_c);
+            p_148840_1_.writeLong(this.field_149328_c);
         }
     }
 
